modules/openlmis-web/src/main/webapp/public/js/distribution/model/epi-inventory.js

```diff
@@ -10,18 +10,50 @@
 
 function EpiInventory(epiInventory) {
   $.extend(true, this, epiInventory);
+
   $(this.lineItems).each(function (index, lineItem) {
     lineItem.existingQuantity = this.existingQuantity || {};
     lineItem.spoiledQuantity = this.spoiledQuantity || {};
   });
 
   var mandatoryFields = ['existingQuantity', 'deliveredQuantity', 'spoiledQuantity'];
+  var nrFields = ['existingQuantity', 'spoiledQuantity'];
 
-  EpiInventory.prototype.setNotRecorded = function () {
-    $(this.lineItems).each(function (index, lineItem) {
-      lineItem.existingQuantity.notRecorded = true;
-      lineItem.spoiledQuantity.notRecorded = true;
+  function init() {
+    var countNotNR = 0;
+    var countNR = 0;
+    $(this.lineItems).each(function (i, lineItem) {
+      $(nrFields).each(function (i, fieldName) {
+        if(isUndefined(lineItem[fieldName]) || lineItem[fieldName].notRecorded === false) {
+            countNotNR++;
+        }
+        else if(lineItem[fieldName].notRecorded === true) {
+            countNR++;
+        }
+      });
     });
+    if(countNR > countNotNR) {
+       this.notRecordedApplied = true;
+    } else {
+       this.notRecordedApplied = false;
+    }
+  }
+
+  init.call(this);
+
+  EpiInventory.prototype.setNotRecorded = function () {
+    if(!this.notRecordedApplied) {
+      $(this.lineItems).each(function (index, lineItem) {
+        lineItem.existingQuantity.notRecorded = true;
+        lineItem.spoiledQuantity.notRecorded = true;
+      });
+    } else {
+       $(this.lineItems).each(function (index, lineItem) {
+         lineItem.existingQuantity.notRecorded = false;
+         lineItem.spoiledQuantity.notRecorded = false;
+       });
+    }
```

Comment:
Can we avoid the need for an if/else by setting values as follows?

     lineItem.existingQuantity.notRecorded = !this.notRecordedApplied;
     lineItem.spoiledQuantity.notRecorded = !this.notRecordedApplied;